lib/internal/repl.js

```diff
@@ -51,19 +53,40 @@ function createRepl(env, cb) {
   }
 
   const repl = REPL.start(opts);
-  if (opts.terminal && env.NODE_REPL_HISTORY_FILE) {
-    return setupHistory(repl, env.NODE_REPL_HISTORY_FILE, cb);
+  if (opts.terminal && env.NODE_REPL_HISTORY !== '') {
+    return setupHistory(repl, env.NODE_REPL_HISTORY,
+                        env.NODE_REPL_HISTORY_FILE, cb);
   }
   repl._historyPrev = _replHistoryMessage;
   cb(null, repl);
 }
 
-function setupHistory(repl, historyPath, ready) {
-  const fs = require('fs');
+function setupHistory(repl, historyPath, oldHistoryPath, ready) {
+  if (!historyPath) {
+    historyPath = path.join(os.homedir(), '.node_repl_history');
```

Comment:
It is possible for `os.homedir()` to `throw`, it would be very uncommon but there are some scenarios beyond `ENOMEM`, `ENOBUFS`, `ENOENT`, `EMFILE` etc. that might be useful to report rather than just crash. I wonder if we should try/catch this and early return `ready()`?